go_modules

```diff
@@ -81,26 +82,32 @@ def get_archive_parameters(args):
 
 
 def basename_from_archive_name(archive_name):
-    basename = re.sub(
-        r"^(?P<service_prefix>_service:[^:]+:)?(?P<basename>.*)\.(?P<extension>obscpio|tar\.[^\.]+)$",
-        r"\g<basename>",
-        archive_name,
-    )
+    if os.path.isdir(archive_name):
```

Comment:
@mcepl this conditional causes the local git clone to be selected as the archive at a higher priority than any configured source archives. This will bypass excludes, subdir and other configurations packagers may have in _service for tar_scm or obs_scm. We'll need to do something differently here to balance the existing majority of packages who use tar_scm or obs_scm, while we prepare for the emerging Git submodule workflow.

Reply:
I am not sure how this code specifically stops `go_modules` from considering those other program parameters. Care to elaborate, please?